gossipsub.go

```diff
@@ -163,21 +163,22 @@ var (
 	GossipSubIWantFollowupTime = 3 * time.Second
 )
 
-// NewGossipSub returns a new PubSub object using GossipSubRouter as the router.
-func NewGossipSub(ctx context.Context, h host.Host, opts ...Option) (*PubSub, error) {
+// NewGossipSubWithProtocols returns a new PubSub object using GossipSubRouter as the router and the protocols specified in ps.
+func NewGossipSubWithProtocols(ctx context.Context, h host.Host, ps []protocol.ID, opts ...Option) (*PubSub, error) {
```

Comment:
this is much better handled with a `WithGossipSubProtocols(protocols, featureCheckFun)` option.